stmhal/dma.h

```diff
@@ -23,83 +23,25 @@
-
-// DMA1 streams
-
-#define DMA_STREAM_I2C1_RX      DMA1_Stream0
-#define DMA_CHANNEL_I2C1_RX                     DMA_CHANNEL_1
-
-#define DMA_STREAM_SPI3_RX      DMA1_Stream2
-#define DMA_CHANNEL_SPI3_RX                     DMA_CHANNEL_0
-
-#define DMA_STREAM_I2C3_RX      DMA1_Stream2
-#define DMA_CHANNEL_I2C3_RX                     DMA_CHANNEL_3
-
-#define DMA_STREAM_I2C2_RX      DMA1_Stream2
-#define DMA_CHANNEL_I2C2_RX                     DMA_CHANNEL_7
-
-#define DMA_STREAM_SPI2_RX      DMA1_Stream3
-#define DMA_CHANNEL_SPI2_RX                     DMA_CHANNEL_0
-
-#define DMA_STREAM_SPI2_TX      DMA1_Stream4
-#define DMA_CHANNEL_SPI2_TX                     DMA_CHANNEL_0
-
-#define DMA_STREAM_I2C3_TX      DMA1_Stream4
-#define DMA_CHANNEL_I2C3_TX                     DMA_CHANNEL_3
-
-#define DMA_STREAM_DAC1         DMA1_Stream5
-#define DMA_CHANNEL_DAC1                        DMA_CHANNEL_7
-
-#define DMA_STREAM_DAC2         DMA1_Stream6
-#define DMA_CHANNEL_DAC2                        DMA_CHANNEL_7
-
-#define DMA_STREAM_SPI3_TX      DMA1_Stream7
-#define DMA_CHANNEL_SPI3_TX                     DMA_CHANNEL_0
-
-#define DMA_STREAM_I2C1_TX      DMA1_Stream7
-#define DMA_CHANNEL_I2C1_TX                     DMA_CHANNEL_1
-
-#define DMA_STREAM_I2C2_TX      DMA1_Stream7
-#define DMA_CHANNEL_I2C2_TX                     DMA_CHANNEL_7
-
-// DMA2 streams
-
-#define DMA_STREAM_SPI1_RX      DMA2_Stream2
-#define DMA_CHANNEL_SPI1_RX                     DMA_CHANNEL_3
-
-#define DMA_STREAM_SPI5_RX      DMA2_Stream3
-#define DMA_CHANNEL_SPI5_RX                     DMA_CHANNEL_2
-
-#define DMA_STREAM_SDIO_RX      DMA2_Stream3
-#define DMA_CHANNEL_SDIO_RX                     DMA_CHANNEL_4
-
-#define DMA_STREAM_SPI4_RX      DMA2_Stream3
-#define DMA_CHANNEL_SPI4_RX                     DMA_CHANNEL_5
-
-#define DMA_STREAM_SPI5_TX      DMA2_Stream4
-#define DMA_CHANNEL_SPI5_TX                     DMA_CHANNEL_2
-
-#define DMA_STREAM_SPI4_TX      DMA2_Stream4
-#define DMA_CHANNEL_SPI4_TX                     DMA_CHANNEL_5
-
-#define DMA_STREAM_SPI6_TX      DMA2_Stream5
-#define DMA_CHANNEL_SPI6_TX                     DMA_CHANNEL_1
-
-#define DMA_STREAM_SPI1_TX      DMA2_Stream5
-#define DMA_CHANNEL_SPI1_TX                     DMA_CHANNEL_3
-
-#define DMA_STREAM_SPI6_RX      DMA2_Stream6
-#define DMA_CHANNEL_SPI6_RX                     DMA_CHANNEL_1
-
-#define DMA_STREAM_SDIO_TX      DMA2_Stream6
-#define DMA_CHANNEL_SDIO_TX                     DMA_CHANNEL_4
+typedef enum {
+    dma_NONE,
+    dma_SPI,
+    dma_QUADSPI,
+    dma_I2C,
+    dma_DAC,
+    dma_ADC,
+    dma_SDIO,
+} periphery_t;
+
+typedef struct _dmaDescr
+{
+    periphery_t     periphery_type;     /* Periphery type */
+    uint32_t        periphery_inst_nr;  /* Instance of Periphery */
+    uint32_t        transfer_direction; /* Transfer direction Periphery to memory or vis-versa */
+} dma_descr_t;
+
+#define DMA_TX_TRANSFER      DMA_MEMORY_TO_PERIPH
+#define DMA_RX_TRANSFER      DMA_PERIPH_TO_MEMORY
```

Comment:
@tobbad why did you declare these 2 macros?  Why not just use DMA_MEMORY_TO_PERIPH and DMA_PERIPH_TO_MEMORY directly?  To me, these are much clearer in their meaning than tx/rx.